core/os/spin_lock.h

```diff
@@ -35,17 +35,46 @@
 
 #include <atomic>
 
+#if !(defined(__arm__) || defined(__aarch64__) || defined(_M_ARM) || defined(_M_ARM64))
```

Comment:
This probably should be the opposite (check for x86 / x86_64) or it will try to use `immintrin` on other non x86 platforms (like RISC-V).